### ISSUE TITLE
Add "clean up" option to "Customizable block shape" 

### ISSUE THREAD

**Issue by sfederici** [ScratchAddons/ScratchAddons#4535]
### Is your request related to a problem? Please describe.

I use the "Customizable block shape" addon. Every time I load a project created by someone else I have to clean up the script areas of all sprites.

### Describe the solution you'd like

Add a "clean up" option to the "Customizable block shape" that would automatically clean up the script areas of all sprites when the project is loaded (using the improved/regular clean up mode selected in the "Developer tools" addon, if enabled) 

### Describe alternatives you've considered

None

**Comment by WorldLanguages** [comment]
I'm not sure if it would be a good idea to trigger changes to the project automatically, but a "clean up all sprites" button could be added and would be useful for more situations, not just this one.

**Comment by sfederici** [comment]
> I'm not sure if it would be a good idea to trigger changes to the project automatically

It is true that it would be automatic, but on explicit request of the user that ticks the option in the "Customizable block shape" addon

> a "clean up all sprites" button could be added and would be useful for more situations, not just this one.

It is true, but, at least for my needs, an automatic beahviour would be a quicker answer to my problem

**Comment by Cool-Scratcher** [comment]
> It is true that it would be automatic, but on explicit request of the user that ticks the option in the "Customizable block shape" addon

It would still be better to have a manual button imo. Maybe there could be another option to make it automatic instead of a button?

**Comment by sfederici** [comment]
> Maybe there could be another option to make it automatic instead of a button?

I don't know. But asking every time would be annoying. 

> It would still be better to have a manual button imo. 

I don't understand why. If I explicitely activated an option to automatically clean up all scripts, why should it be a problem having an auotmatic clean up? Maybe there could be two options, one for automatic clean up (that is what I would really like) and one for clean up on request (buttom, other, etc)?

**Comment by WorldLanguages** [comment]
> I don't understand why. If I explicitely activated an option to automatically clean up all scripts, why should it be a problem having an auotmatic clean up?

I don't think we have any features yet that automatically make changes to the project after some event. It's something I don't think users expect, even if we make it a setting.
Also, I'm not sure if it would be feasible to make it possible to undo the clean up with Ctrl+Z in all sprites at the same time. At a minimum, an automatically triggered action should be undoable.

**Comment by sfederici** [comment]
> I don't think we have any features yet that automatically make changes to the project after some event. It's something I don't think users expect, even if we make it a setting.

I see. So this is new. But even new is not this bad. Maybe we could have a red sign warning clarifying this option is "dangerous" (as other already existing options)

> I'm not sure if it would be feasible to make it possible to undo the clean up with Ctrl+Z in all sprites at the same time. At a minimum, an automatically triggered action should be undoable.

This is not true, as least for this very addon. Indeed, as this is automatically done at load time, if I realize that I don't need it I can just close the project, disable the option, and open the project again. 

Just to make the user -they could have forgotten of having enabled the option, or they could have not fully understood what this was for- aware that the option is active, at load time we could have a dialog with a warning that remembers that the blocks have been rearranged due to the addon. I could live with that.
